src/main/java/com/datastax/oss/protocol/internal/request/AuthResponse.java

```diff
@@ -43,7 +49,16 @@ public Codec(int protocolVersion) {
     @Override
     public <B> void encode(B dest, Message message, PrimitiveCodec<B> encoder) {
       AuthResponse authResponse = (AuthResponse) message;
-      encoder.writeBytes(authResponse.token, dest);
+      ByteBuffer token = authResponse.token;
+      token.mark();
+      encoder.writeBytes(token, dest);
+
+      if (!token.isReadOnly()) {
```

Comment:
How about isolating this code in a utility method, e.g. in the `Bytes` class? This would make the intent even more clear.